### ISSUE TITLE
xcodes() stopped working -- no changes or updates

### ISSUE THREAD

**Issue by FeelHippo** [fastlane/fastlane#21992]
- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
`xcodes()` has seemingly stopped working overnight. We have been using it in a CI lane for quite some time, and up until last week it was working. We have not made any changes to the lane, nor we have modified the CI workflow or added/removed any dependencies. 

##### Command executed
```
xcodes(
      version: "14.1",
      select_for_current_build_only: true,
)
```
##### Complete output when running fastlane, including the stack trace and command used

- `xcodes( version:"14.1", select_for_current_build_only: true)` -- LAST WEEK
```
[08:32:36]: --------------------
[08:32:36]: --- Step: xcodes ---
[08:32:36]: --------------------
[08:32:36]: Running xcodes version 1.4.1
[08:32:36]: $ /usr/local/bin/xcodes installed '14.1'
[08:32:36]: ▸ /Applications/Xcode_14.1.app
[08:32:36]: Setting Xcode version '14.1' at '/Applications/Xcode_14.1.app' for all build steps
```
- `xcodes( version:"14.1", select_for_current_build_only: true)` -- TODAY
```
[13:27:44]: --------------------
[13:27:44]: --- Step: xcodes ---
[13:27:44]: --------------------
[13:27:44]: Running xcodes version 1.4.1
[13:27:44]: $ /opt/homebrew/bin/xcodes installed '14.1'
[13:27:44]: ▸ 14.1 is not installed.
[13:27:44]: Exit status of command '/opt/homebrew/bin/xcodes installed '14.1'' was 1 instead of 0.
14.1 is not installed.
```

Please note the different output: `/usr/local/bin/xcodes installed '14.1'` (works), `/opt/homebrew/bin/xcodes installed '14.1'` (doesn't). I thought it might be due to Xcodes, as the [migration docs](https://github.com/xcpretty/xcode-install/blob/master/MIGRATION.md#migrating-fastlane-actions-that-depend-on-the-xcode-install-gem-to-use-xcodes) state that `make sure you have it installed in your environment`, so the CI workflow installs it right before getting to the lane: `brew install xcodesorg/made/xcodes`. Same result. 

### Environment

<details>
  <pre> <details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                        |
| --------------------------- | ------------------------------------------------------------ |
| OS                          | 13.3                                                         |
| Ruby                        | 3.3.0                                                        |
| Bundler?                    | false                                                        |
| Git                         | git version 2.39.2 (Apple Git-143)                           |
| Installation Source         | /opt/homebrew/Cellar/fastlane/2.219.0_2/libexec/bin/fastlane |
| Host                        | macOS 13.3 (22E252)                                          |
| Ruby Lib Dir                | /opt/homebrew/Cellar/ruby/3.3.0/lib                          |
| OpenSSL Version             | OpenSSL 3.2.0 23 Nov 2023                                    |
| Is contained                | false                                                        |
| Is homebrew                 | true                                                         |
| Is installed via Fabric.app | false                                                        |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                  |
| Xcode Version               | 14.3.1                                                       |
| Swift Version               | 5.8.1                                                        |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.220.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| error_highlight                 | 0.6.0        |
| did_you_mean                    | 1.6.3        |
| syntax_suggest                  | 2.0.0        |
| addressable                     | 2.8.6        |
| babosa                          | 1.0.4        |
| bundler                         | 2.5.4        |
| colored                         | 1.2          |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| dotenv                          | 2.8.1        |
| emoji_regex                     | 3.2.3        |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-httpclient              | 1.0.1        |
| faraday-multipart               | 1.0.4        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.2.0        |
| faraday-patron                  | 1.0.0        |
| faraday-rack                    | 1.0.0        |
| faraday-retry                   | 1.0.3        |
| ruby2_keywords                  | 0.0.5        |
| faraday                         | 1.10.3       |
| faraday_middleware              | 1.2.0        |
| domain_name                     | 0.6.20240107 |
| http-cookie                     | 1.0.5        |
| faraday-cookie_jar              | 0.0.7        |
| gh_inspector                    | 1.1.3        |
| google-apis-androidpublisher_v3 | 0.54.0       |
| google-apis-playcustomapp_v1    | 0.13.0       |
| google-cloud-env                | 1.6.0        |
| google-apis-iamcredentials_v1   | 0.17.0       |
| google-apis-storage_v1          | 0.31.0       |
| multi_json                      | 1.15.0       |
| os                              | 1.1.4        |
| googleauth                      | 1.8.1        |
| digest-crc                      | 0.6.5        |
| mini_mime                       | 1.1.5        |
| google-cloud-storage            | 1.47.0       |
| mini_magick                     | 4.12.0       |
| naturally                       | 2.2.1        |
| optparse                        | 0.4.0        |
| plist                           | 3.7.1        |
| rubyzip                         | 2.3.2        |
| security                        | 0.1.5        |
| simctl                          | 1.6.10       |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 2.5.0        |
| terminal-table                  | 3.0.2        |
| tty-screen                      | 0.8.2        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| word_wrap                       | 1.0.0        |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| xcpretty-travis-formatter       | 1.0.1        |
| set                             | 1.1.0        |
| public_suffix                   | 5.0.5        |
| artifactory                     | 3.0.17       |
| aws-eventstream                 | 1.3.0        |
| aws-sigv4                       | 1.8.0        |
| aws-partitions                  | 1.907.0      |
| jmespath                        | 1.6.2        |
| aws-sdk-core                    | 3.191.6      |
| aws-sdk-kms                     | 1.78.0       |
| aws-sdk-s3                      | 1.146.1      |
| base64                          | 0.2.0        |
| nkf                             | 0.2.0        |
| rexml                           | 3.2.6        |
| CFPropertyList                  | 3.0.7        |
| excon                           | 0.110.0      |
| multipart-post                  | 2.4.0        |
| fastimage                       | 2.3.1        |
| httpclient                      | 2.8.3        |
| jwt                             | 2.8.1        |
| signet                          | 0.19.0       |
| retriable                       | 3.1.2        |
| trailblazer-option              | 0.1.2        |
| declarative                     | 0.0.20       |
| uber                            | 0.1.0        |
| representable                   | 3.2.0        |
| google-apis-core                | 0.11.3       |
| google-cloud-errors             | 1.4.0        |
| google-cloud-core               | 1.7.0        |
| rake                            | 13.2.0       |
| json                            | 2.7.2        |
| nanaimo                         | 0.3.0        |
| colored2                        | 3.1.2        |
| claide                          | 1.1.0        |
| atomos                          | 0.1.3        |
| xcodeproj                       | 1.24.0       |
| forwardable                     | 1.3.3        |
| logger                          | 1.6.0        |
| pathname                        | 0.3.0        |
| shellwords                      | 0.2.0        |
| cgi                             | 0.4.1        |
| date                            | 3.3.4        |
| timeout                         | 0.4.1        |
| securerandom                    | 0.3.1        |
| uri                             | 0.13.0       |
| openssl                         | 3.2.0        |
| digest                          | 3.1.1        |
| ipaddr                          | 1.2.6        |
| resolv                          | 0.3.0        |
| time                            | 0.3.0        |
| stringio                        | 3.1.0        |
| open-uri                        | 0.4.1        |
| mutex_m                         | 0.2.0        |
| net-http                        | 0.4.0        |
| net-protocol                    | 0.2.2        |
| english                         | 0.8.0        |
| erb                             | 4.0.3        |
| abbrev                          | 0.1.2        |
| tempfile                        | 0.2.1        |
| delegate                        | 0.3.1        |
| fileutils                       | 1.7.2        |
| tmpdir                          | 0.2.0        |
| singleton                       | 0.2.0        |
| open3                           | 0.2.1        |
| prettyprint                     | 0.2.0        |
| pp                              | 0.5.0        |
| find                            | 0.2.0        |
| ostruct                         | 0.6.0        |
| yaml                            | 0.3.0        |
| psych                           | 5.1.2        |
</details>

*generated on:* **2024-04-29**
</details> </pre>
</details>

**Comment by rileywrites** [comment]
Same error/output here, for Xcode 15:

```
[17:13:05]: --------------------
[17:13:05]: --- Step: xcodes ---
[17:13:05]: --------------------
[17:13:05]: Running xcodes version 1.4.1
[17:13:05]: $ /opt/homebrew/bin/xcodes installed '15'
[17:13:05]: ▸ 15.0 is not installed.
[17:13:05]: Exit status of command '/opt/homebrew/bin/xcodes installed '15'' was 1 instead of 0.
15.0 is not installed.
```

i've gone back to using `xcversion()` for now, which works fine, but i'd prefer to use the non-deprecated action ^-^

**Comment by FeelHippo** [comment]
I did try to revert to `xcversion()` as well, though my pipeline failed (kind of expected, unless you specify an older Fastlane version in Gemfile):
```
[11:06:50]: -----------------------
[11:06:50]: --- Step: xcversion ---
[11:06:50]: -----------------------
==========================================
This action (xcversion) is deprecated
The xcode-install gem, which this action depends on, has been sunset. Please migrate to "xcodes" (https://docs.fastlane.tools/actions/xcodes 🔗). You can find a migration guide here: "xcpretty/xcode-install/MIGRATION.md" (https://github.com/xcpretty/xcode-install/blob/master/MIGRATION.md 🔗)
==========================================
```

**Comment by FeelHippo** [comment]
Issue fixed by using xCode 15.1.0:

My CI looks like this:

```
...
    steps:
      - name: Install Bundler
        run: gem install bundler
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:
          node-version: 16
      - uses: maxim-lobanov/setup-xcode@v1
        with:
          xcode-version: '15.1.0'
...
      - name: Archive, sign and deploy using Fastlane
        run: bundle exec fastlane deploy_updraft
        working-directory: ./ios
        env:
          XCVERSION: '15.1.0'
```

and in my Fastfile:

```
...
xcodes(
      version: ENV["XCVERSION"],
      select_for_current_build_only: true,
)
```

**Comment by lacostej** [comment]
Were you building on github? If so, it might be because github had removed that particular version from your image?

Anyway this looks old, and self-solved, so closing.
